porch/repository/pkg/git/git.go

```diff
@@ -643,3 +630,106 @@ func (r *gitRepository) getRepo() (string, error) {
 
 	return origin.Config().URLs[0], nil
 }
+
+func (r *gitRepository) update(ctx context.Context) error {
+	auth, err := r.getAuthMethod(ctx)
+	if err != nil {
+		return err
+	}
+
+	// Fetch
+	switch err := r.repo.Fetch(&gogit.FetchOptions{
+		RemoteName: originName,
+		Auth:       auth,
+		Tags:       gogit.AllTags,
+	}); err {
+	case nil: // OK
+	case gogit.NoErrAlreadyUpToDate:
```

Comment:
Well that's a frustrating API :-)

Reply:
yes :) Errors used quite a bit to signal success, and some of the errors are are named as errors (`ErrEmptyRemoteRepository`)